src/core/namespace.c

```diff
@@ -1048,11 +1091,9 @@ int setup_namespace(
                 if (r < 0)
                         goto finish;
 
-                if (namespace_info_mount_apivfs(root_directory, ns_info)) {
-                        r = append_static_mounts(&m, apivfs_table, ELEMENTSOF(apivfs_table), ns_info->ignore_protect_paths);
-                        if (r < 0)
-                                goto finish;
-                }
+                r = append_mount_apivfs(&m, root_directory, ns_info, ns_info->ignore_protect_paths);
+                if (r < 0)
+                        goto finish;
```

Comment:
so, i am not sure I like this specific change, simply as a matter of being unnecessarily fine-grained. I am not sure it's worth optimizing in this area, I much rather would say that /proc, /proc and /dev being around is the absolute baseline of APIs to expect if you are a Linux C program. I mean, MountAPIVFS= is an option instead of the default primarily to keep compat with any status quo ante, not because it was really a good idea to make /proc and friends optional...

I think there's benefit in simplicity, and being able to simply say in the man pages that MountAPIVFS= is implied if ProtectControlGroups= or ProtectKernelTunables= is set is pretty simply. Having to say that parts of MountAPIVFS= are implied, and then also clarifying which ones doesn't appear a net benefit to me...

I'd prefer if this patch could be dropped, I must say..